### PR TITLE
Add shape documentation for CosineEmbeddingLoss

### LINKED ISSUE #52732: The shape of label tensor for CosineEmbeddingLoss should be specified in the document
## 📚 Documentation

I'm now using torch 1.7.1. And in the [document for CosineEmbeddingLoss](https://pytorch.org/docs/stable/generated/torch.nn.CosineEmbeddingLoss.html#torch.nn.CosineEmbeddingLoss), it should be mentioned that the label tensor **y** has to be **a 1D mini-batch Tensor**. Otherwise, even though the label tensor contains  only -1 and 1, the loss value would be incorrect if the tensor is a 2D mini-batch Tensor.

For example:
```python
import torch
import torch.nn as nn

torch.manual_seed(0)

a = torch.randn(16, 5)
b = torch.randn(16, 5)
labels = torch.empty(16).bernoulli_().mul_(2).sub_(1)

loss_fn = nn.CosineEmbeddingLoss()

print(loss_fn(a, b, labels))  # 0.4538
print(loss_fn(a, b, labels.unsqueeze(1)))  # 0.4810
```

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @anjali411 @brianjo @mruberry @albanD

### PR DESCRIPTION
**Pull request by jbschlosser**
Fixes #52732

### CODE REVIEW

File: torch/nn/modules/loss.py [hunk 1]
```diff
@@ -1203,6 +1203,12 @@ class CosineEmbeddingLoss(_Loss):
             elements in the output, ``'sum'``: the output will be summed. Note: :attr:`size_average`
             and :attr:`reduce` are in the process of being deprecated, and in the meantime,
             specifying either of those two args will override :attr:`reduction`. Default: ``'mean'``
+
+    Shape:
+        - Input1: :math:`(N, D)`, where `N` is the batch size and `D` is the embedding dimension.
+        - Input2: :math:`(N, D)`, same shape as Input1.
+        - Target: :math:`(N)`.
+        - Output: scalar. If :attr:`reduction` is ``'none'``, then :math:`(N)`.
```
Comment (albanD) [inline torch/nn/modules/loss.py:1211]: It's not super clear that the "scalar" is the else case here?
:math:`(N)` if :attr:`reduction` is ``'none'`` otherwise scalar

**Review by albanD — APPROVED** [review]
nit phrasing, good otherwise.
